src/chigame/users/models.py

```diff
@@ -33,3 +33,57 @@ def get_absolute_url(self) -> str:
 
         """
         return reverse("users:detail", kwargs={"pk": self.id})
+
+
+class UserProfile(models.Model):
+    """
+    User profile.
+
+    We separate the profile from the User model, to allow for users
+    that don't need a profile on the website (e.g., API-only users)
+    """
+
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    display_name = models.TextField()
+    bio = models.TextField(blank=True)
+    friends = models.ManyToManyField(User, related_name="friendship", blank=True)
+    date_joined = models.DateTimeField(auto_now_add=True)
+
+    @classmethod
+    def get_or_create_profile(cls, user: User) -> "UserProfile":
+        profile, created = cls.objects.get_or_create(user=user)
+        return profile
+
+
+class FriendInvitation(models.Model):
+    """
+    An invitation from a User to another User, requesting that they become friends.
+    """
+
+    sender = models.ForeignKey(User, related_name="sent_friend_invitations", on_delete=models.CASCADE)
+    receiver = models.ForeignKey(User, related_name="received_friend_invitations", on_delete=models.CASCADE)
+    accepted = models.BooleanField(default=False)
+    timestamp = models.DateTimeField(auto_now_add=True)
+
+
+class FriendGroup(models.Model):
```

Comment:
Is a friend group necessarily a group of people that are all mutually Friends (as in have accepted FriendInvitation entries)?

Or is a FriendGroup any collection of users.

If it's the latter, then I feel the name could be misleading, might want to be changed to Group or PlayerGroup

Reply:
We had clarified this in the Feature Wishlist, but hadn't changed the corresponding code. I've pushed an update that changes all references of "friend groups" to just "groups"